external/js/cothority/src/byzcoin/byzcoin-rpc.ts

```diff
@@ -30,6 +30,9 @@ export const currentVersion = 2;
 const CONFIG_INSTANCE_ID = Buffer.alloc(32, 0);
 
 export default class ByzCoinRPC implements ICounterUpdater {
+
+    static staticCounters: Map<string, Map<string, Long>> = new Map<string, Map<string, Long>>();
```

Comment:
Why having it static? Does the rpc get copied a lot or only referenced?

Reply:
Also, why using `Map<(string,string),Long>` instead of using direclty the `InstanceIDs`?

Reply:
Static: there are multiple instances of `ByzCoinRPC` in use, at least in the `mobile`, but probably also in the `webapp`.

`Map<InstanceID` - IIRC I did that in the beginning, but it didn't work from time to time. The problem is that `Map.set(Buffer.from("123", 1)).get(Buffer.from("123")` returns `undefined`, as far as I remember. But with strings it works...

See https://runkit.com/ineiti/5d9e0424577a31001a6341d2

Reply:
Ho, okay, it turns out that [JS Map doesn't use structural equality, only reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), so `Object` can't be used as key. To work around that, we can use [immutable.js](https://immutable-js.github.io/immutable-js/) to have better collections, key equality tested using `equals` for Map/List/Seq. which is kinda the standard by default, even if badly used by stdlib.